doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,90 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically. For example, solving $x^2 = 4$ yields $x \in \{-2,2\}$.
+
+There are two high-level functions to solve equations, [`solve`](#) and [`solveset`](#). Here are their advantages and disadvantages: ... *table?*
+
+## Using {func}`~.solve`
+
+{func}`~.solve`
+- is best for solving equations that...
+- produces outputs in the format of a Python list.
+
+You can solve an equation using {func}`~.solve` in several ways.
+
+1. Use the fact that any expression not in an `Eq` (equation) is automatically assumed to equal 0 by the solving functions. You can rearrange the equation $x^2 = 4$ to $x^2 - 4 = 0$, and {func}`~.solve` that expression. This approach is convenient if you are interactively solving an equation which can easily be rearranged to $expression = 0$.
+
+```
+>>> from sympy import solve
+>>> from sympy.abc import x
+>>> solve(x**2 - 4)
+[-2, 2]
+```
+
+2. Put your equation into `Eq` form, then apply {func}`~.solve` to the `Eq`. This approach is convenient if you are interactively solving an equation which cannot easily be rearranged to $expression = 0$.
+
+```
+>>> from sympy import solve, Eq
+>>> from sympy.abc import x
+>>> eqn = Eq(x**2, 4)
+>>> solve(eqn)
+[-2, 2]
+```
+
+3. Parse a string representing the equation into a form that SymPy can understand, then apply {func}`~.solve` to the parsed expression. 
+
+*But this doesn't work for equality; does for inequality. Is there some way to do this for equality?*
```

Comment:
```
In [11]: solve(parse_expr("Eq(x**2, 1)"))
Out[11]: [-1, 1]
```

Reply:
Thanks, that worked.

I also want to demonstrate parsing a string like "x**2 = y" -- that's the kind of expression that I originally used SymPy for, and I think a fairly likely use case. I tried using transformations=(convert_equals_signs) as @asmeurer suggested, but I'm having problems using it, or even importing transformations, as described in the [Parsing documentation](https://docs.sympy.org/dev/modules/parsing.html?highlight=parse_expr#sympy.parsing.sympy_parser.parse_expr). Any suggestions?

```
>>> solve(parse_expr("x**2 == y"), transformations=(convert_equals_signs))
Traceback (most recent call last):
  File "<pyshell#5>", line 1, in <module>
    solve(parse_expr("x**2 == y"), transformations=(convert_equals_signs))
NameError: name 'convert_equals_signs' is not defined
>>> transformations = (convert_equals_signs)
Traceback (most recent call last):
  File "<pyshell#6>", line 1, in <module>
    transformations = (convert_equals_signs)
NameError: name 'convert_equals_signs' is not defined
>>> from sympy.parsing.sympy_parser import transformations
Traceback (most recent call last):
  File "<pyshell#7>", line 1, in <module>
    from sympy.parsing.sympy_parser import transformations
ImportError: cannot import name 'transformations' from 'sympy.parsing.sympy_parser' (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/sympy/parsing/sympy_parser.py)
```

Reply:
Maybe something wrong with your installation. I can import it fine.

Reply:
Hmm, weird. It works fine when I run it in Visual Studio Code. It didn't work in in IDLE, so maybe something was wrong with how SymPy got imported there.

Reply:
Some progress: I can import transformations (I think it was a venv or conda issue).

In the following code, `parsed1` and `parsed2` work fine, but either of the last two lines, where I try to parse an expression containing an equals sign using `convert_equals_signs`, throws the exception `name 'x' is not defined`. I'm confused because I imported x from sympy.abc.

```
from sympy import parse_expr, solve
from sympy.abc import x
from sympy.parsing.sympy_parser import T, convert_equals_signs, transformations

parsed1 = parse_expr("2x", transformations=T[:6])
print(parsed1)
parsed2 = parse_expr("2 * x")
print(parsed2)
s = solve(parsed2, x)
print(s)

parsed3 = parse_expr("2 * x = 1", transformations=[convert_equals_signs])
parsed4 = parse_expr("2 * x = 1", transformations=T[9])
```

Reply:
You have to also include the standard_transformations (the default value of transformations). Something like `transformations=default_transformations + (convert_equals_signs,)` This whole thing is somewhat advanced, at least in its current state.

Reply:
Or you can use `transformations=T[1,9]` for both cases in parsed3-6.